scalekit/core.py

```
@@ -164,6 +164,8 @@ def grpc_exec(
             return resp
         except grpc.RpcError as exp:
             if exp.code() == grpc.StatusCode.UNAUTHENTICATED:
+                if retry <= 0:
+                    raise ScalekitServerException.promote(exp)
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)